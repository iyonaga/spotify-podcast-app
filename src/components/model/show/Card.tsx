import Image from 'next/image';
import Link from 'next/link';

interface Props {
  show: SpotifyApi.ShowObjectSimplified;
}

const Card: React.VFC<Props> = ({ show }) => {
  return (
    <div className="w-full bg-[#252332] hover:bg-[#2a2838] rounded-[5px] transition">
      <Link href={`/show/${show.id}`}>
        <a className="flex flex-col p-[15px] h-full">
          <Image
            className="rounded-[9px]"
            src={show.images[1].url}
            width={show.images[1].width}
            height={show.images[1].height}
            alt={show.name}
            layout="responsive"
          />
          <p className="mt-[9px] text-16px font-bold">{show.name}</p>
          <p className="mt-[6px] text-13px">{show.publisher}</p>
        </a>
      </Link>
    </div>
  );
};

export default Card;
