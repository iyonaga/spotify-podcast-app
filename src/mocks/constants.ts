const API_BASE_URL = 'https://api.spotify.com/v1';

export const ENDPOINTS = {
  // shows
  GET_SHOW_EPISODES: `${API_BASE_URL}/shows/:id/episodes`,
  SAVE_SHOWS: `${API_BASE_URL}/me/shows`,
  REMOVE_SAVED_SHOWS: `${API_BASE_URL}/me/shows`,
  CHECK_SAVED_SHOWS: `${API_BASE_URL}/me/shows/contains`,

  // episodes
  GET_SAVED_EPISODES: `${API_BASE_URL}/me/episodes`,
  SAVE_EPISODES: `${API_BASE_URL}/me/episodes`,
  REMOVE_SAVED_EPISODES: `${API_BASE_URL}/me/episodes`,
  CHECK_SAVED_EPISODES: `${API_BASE_URL}/me/episodes/contains`,
};
