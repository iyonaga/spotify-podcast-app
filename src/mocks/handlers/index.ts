import { handlers as episodesHandlers } from './episodes';
import { handlers as searchHandlers } from './search';
import { handlers as showsHandlers } from './shows';

export const handlers = [
  ...showsHandlers,
  ...episodesHandlers,
  ...searchHandlers,
];
