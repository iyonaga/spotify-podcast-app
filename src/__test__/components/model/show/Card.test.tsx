import { render } from '@testing-library/react';
import Card from '@/components/model/show/Card';
import { showsMock } from '@/mocks/showsMock';

describe('components/model/show/Card.tsx', () => {
  test('スナップショットが一致する', () => {
    const { asFragment } = render(<Card show={showsMock[0]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
