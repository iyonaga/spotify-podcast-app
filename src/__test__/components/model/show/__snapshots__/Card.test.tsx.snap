// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/model/show/Card.tsx スナップショットが一致する 1`] = `
<DocumentFragment>
  <div
    class="w-full bg-[#252332] hover:bg-[#2a2838] rounded-[5px] transition"
  >
    <a
      class="flex flex-col p-[15px] h-full"
      href="/show/6F34Z6K3gaI1iLiRdXhdGt"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
        />
        <img
          alt="英語で雑談！Kevin’s English Room Podcast"
          class="rounded-[9px]"
          data-nimg="responsive"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <p
        class="mt-[9px] text-16px font-bold"
      >
        英語で雑談！Kevin’s English Room Podcast
      </p>
      <p
        class="mt-[6px] text-13px"
      >
        ケビン (Kevin's English Room)
      </p>
    </a>
  </div>
</DocumentFragment>
`;
